File: Samples/visio_managed_code_interop_vs2010/VS1010_CSharp_Samples/shape_getformulas.cs
using System.Collections.Generic;
using IVisio = Microsoft.Office.Interop.Visio;
using VisioInterop;
public static partial class VS2010_CSharp_Samples
{
    public static void Shape_GetFormulas(IVisio.Document doc)
    {
        var page = VisioInterop.Util.CreateStandardPage(doc, "SGF");
        var shape = VisioInterop.Util.CreateStandardShape(page);
        var request = VisioInterop.Util.Create_ShapeGetFormulas_Request();

        // MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        var SRCStream = new short[request.Length * 3];
        for (int i = 0; i < request.Length; i++)
        {
            SRCStream.Set3(i,request[i].CellSRC.Section, request[i].CellSRC.Row, request[i].CellSRC.Cell);
        }

        // EXECUTE THE REQUEST
        System.Array formulas_sa;
        shape.GetFormulasU(SRCStream, out formulas_sa);

        // MAP OUTPUT BACK TO SOMETHING USEFUL 
        var formulas = new string[formulas_sa.Length];
        formulas_sa.CopyTo(formulas, 0);

        // DISPLAY THE INFORMATION
        shape.Text = string.Format("Formulas={0},{1}", formulas[0], formulas[1]);
    }
}

File: Samples/visio_managed_code_interop_vs2010/Visio2010_VB_Samples/Shape_GetFormulas.vb
Imports IVisio = Microsoft.Office.Interop.Visio


Partial Public Class VS2010_VB_Samples
    Shared Sub Shape_GetFormulas (ByVal doc As Microsoft.Office.Interop.Visio.Document)

        Dim page = Util.CreateStandardPage(doc, "SGF")
        Dim shape = Util.CreateStandardShape(page)

        ' CREATE REQUEST
        Dim request = {New With { _
                Key .Section = CShort (IVisio.VisSectionIndices.visSectionObject), _
                Key .Row = CShort (IVisio.VisRowIndices.visRowXFormOut), _
                Key .Cell = CShort (IVisio.VisCellIndices.visXFormWidth) _
                }, New With { _
                Key .Section = CShort (IVisio.VisSectionIndices.visSectionObject), _
                Key .Row = CShort (IVisio.VisRowIndices.visRowXFormOut), _
                Key .Cell = CShort (IVisio.VisCellIndices.visXFormHeight) _
                }}

        ' MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        Dim SRCStream = New Short(request.Length*3 - 1) {}
        For i As Integer = 0 To request.Length - 1
            SRCStream ((i*3) + 0) = request (i).Section
            SRCStream ((i*3) + 1) = request (i).Row
            SRCStream ((i*3) + 2) = request (i).Cell
        Next

        ' EXECUTE THE REQUEST
        Dim formulas_sa As System.Array = Nothing
        shape.GetFormulasU (SRCStream, formulas_sa)

        ' MAP OUTPUT BACK TO SOMETHING USEFUL 
        Dim formulas = New String(formulas_sa.Length - 1) {}
        formulas_sa.CopyTo (formulas, 0)

        ' DISPLAY THE INFORMATION
        shape.Text = String.Format ("Formulas={0},{1}", formulas (0), formulas (1))
    End Sub
End Class

